dip-0027.md

```diff
@@ -0,0 +1,163 @@
+
+Eventually, an Identity may want to withdraw their credits from Platform to regain access to the underlying locked Dash. Any Identity type can issue a withdrawal request to Dash Platform. If the Identity’s balance is sufficient, this will result in the formation of a transaction signed by a quorum and valid on the Core chain. This transaction unlocks the requested amount of Dash from the credit pool and sends it to the provided Dash address.
+
+## Asset Locking
+
+Dash Core participates in the credit creation process by locking Dash in the credit pool with an Asset Lock transaction.
+
+### Asset Lock Transaction
+
+The Asset Lock transaction (AssetLockTx) is a DIP-2-based special transaction that gives Platform the information required to assign credits to Identities. This special transaction is type 8.
+
+The format of the Asset Lock special transaction payload:
+
+| **Name** | **Type** | **Size**  | **Description** |
+| - | - | - | - |
+| version | uint8_t | 1 | Asset Lock transaction version number |
+| count | uint8_t | 1 | The number of accounts funded via this Asset Lock |
+| credit_outputs | vec\<TxOut> | 28 * count| The hash of the public key(s) used to claim credits on Dash Platform, as well as the amount of Dash in duffs transferred into credits for this public key hash |
+
+Each Asset Lock transaction must create an unspendable output by using an OP_RETURN as the first script. The following criteria apply to the OP_RETURN output:
+
+* It must not include any data
+* Its value is the amount of Dash to be locked
+* The sum of all TxOut amounts must equal the value of the output
+* It must be the only output in the transaction using an OP_RETURN
+
+### Proof of Finality
+
+It is important to note that Dash Core transactions achieve finality through ChainLocks and InstantSend locks. Therefore, these locks can be used to prove that value has been added to the credit pool and is ready for Platform use. For example, Platform can safely increase an Identity’s balance without concerns that the related Asset Lock transaction will be double-spent.
+
+There are two ways to prove transaction finality: either by including the InstantSend lock of a recent transaction or by referencing the chain height at which a transaction was included in a ChainLocked block. Typically, it is preferable to use the InstantSend lock for proof, but a ChainLock can be used if an InstantSend lock is not available.
+
+## Asset Unlocking
+
+Withdrawing locked credit pool funds back to Dash is done by creating a Core chain unlock transaction signed by an LLMQ_100_67 quorum. This transaction will release Dash from the credit pool while reducing its balance by an equivalent amount.
```

Comment:
Let's mention that this quorum is dedicated to platform validators. There are other types that we use on testnet/devnet. A common property is that they are all for platform validators.

Reply:
> while reducing its balance by an equivalent amount.
I think this part must be clarified. The transaction itself directly doesn't reduce a balance (it's not clear what exactly balance). There is a state transition that reduces the corresponding identity balance and creates an Asset Unlock Transaction that is signed by validators and then broadcasted to Core.

Reply:
> > while reducing its balance by an equivalent amount.
> 
> I think this part needs clarification. A transaction itself directly doesn't reduce any balance (it's not clear what exactly balance we are pointing to). There is a state transition that reduces the corresponding identity balance and creates an Asset Unlock Transaction that is signed by validators and then broadcasted to Core.

I clarified this in https://github.com/dashpay/dips/pull/152/commits/cc9c5a853bd3be23b102737cff8af256f20a7e22. I was intending to refer to the credit pool's balance, not the identity (that part would get covered by the related doc that will discuss the platform side of things). I tried to keep platform-related details out of this doc or generic as much as possible.